craft_parts/executor/environment.py

```diff
@@ -63,16 +59,22 @@ def generate_step_environment(
 
         print("# Environment", file=run_environment)
 
-        print("## Part Environment", file=run_environment)
-        for key, val in our_build_environment.items():
+        print("## Part environment", file=run_environment)
+        for key, val in parts_environment.items():
             print(f'export {key}="{val}"', file=run_environment)
 
-        print("## Plugin Environment", file=run_environment)
+        print("## Plugin environment", file=run_environment)
         for key, val in plugin_environment.items():
             print(f'export {key}="{val}"', file=run_environment)
 
-        print("## User Environment", file=run_environment)
-        for env in user_build_environment:
+        print("## Application environment", file=run_environment)
```

Comment:
application environment should come before part and plugin as these could reference the application provided variables